n26/api.py

```diff
@@ -265,37 +282,51 @@ def get_token(self):
 
         :return: the access token
         """
-        if not self._validate_token(self._token_data):
-            if REFRESH_TOKEN_KEY in self._token_data:
-                refresh_token = self._token_data[REFRESH_TOKEN_KEY]
-                self._token_data = self._refresh_token(refresh_token)
+        token_data = self.token_data
+        if not self._validate_token(token_data):
+            if REFRESH_TOKEN_KEY in token_data:
+                refresh_token = token_data[REFRESH_TOKEN_KEY]
+                token_data = self._refresh_token(refresh_token)
             else:
-                self._token_data = self._request_token(self.config.username, self.config.password)
+                token_data = self._request_token(self.config.username, self.config.password)
 
             # add expiration time to expiration in _validate_token()
-            self._token_data[EXPIRATION_TIME_KEY] = time.time() + self._token_data["expires_in"]
+            token_data[EXPIRATION_TIME_KEY] = time.time() + token_data["expires_in"]
 
         # if it's still not valid, raise an exception
-        if not self._validate_token(self._token_data):
+        if not self._validate_token(token_data):
             raise PermissionError("Unable to request authentication token")
 
-        return self._token_data[ACCESS_TOKEN_KEY]
+        # save token data
+        self.token_data = token_data
+        return token_data[ACCESS_TOKEN_KEY]
 
-    @staticmethod
-    def _request_token(username: str, password: str):
+    def _request_token(self, username: str, password: str):
         """
         Request an authentication token from the server
         :return: the token or None if the response did not contain a token
         """
+        mfa_token = self._initiate_authentication_flow(username, password)
+        self._request_mfa_approval(mfa_token)
+        return self._complete_authentication_flow(mfa_token)
+
+    @staticmethod
+    def _initiate_authentication_flow(username: str, password: str) -> str:
         values_token = {
-            'grant_type': 'password',
-            'username': username,
-            'password': password
+            "grant_type": GRANT_TYPE_PASSWORD,
+            "username": username,
+            "password": password
         }
-
-        response = requests.post(BASE_URL + '/oauth/token', data=values_token, headers=BASIC_AUTH_HEADERS)
-        response.raise_for_status()
-        return response.json()
+        # TODO: Seems like the user-agent is not necessary but might be a good idea anyway
```

Comment:
I'm not sure if we want this, I don't think it is needed, at least not at the moment.

Reply:
I will leave this TODO in the code for now, we can think about it again if it becomes a problem.